src/g1.rs

```diff
@@ -148,52 +264,42 @@ where
     where
         I: Iterator<Item = T>,
     {
-        iter.fold(Self::zero(), |acc, item| acc + item.borrow())
+        iter.fold(Self::identity(), |acc, item| acc + item.borrow())
     }
 }
 
-impl_binops_additive_mixed!(G1Projective, G1Affine, groupy::CurveProjective);
-impl_binops_additive_specify_output!(G1Affine, G1Projective, G1Projective);
+impl ConditionallySelectable for G1Affine {
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        G1Affine(blst_p1_affine {
+            x: Fp::conditional_select(&a.x(), &b.x(), choice).0,
+            y: Fp::conditional_select(&a.y(), &b.y(), choice).0,
+        })
+    }
+}
 
-impl groupy::CurveAffine for G1Affine {
-    type Engine = crate::Bls12;
-    type Scalar = Scalar;
-    type Base = Fp;
-    type Projective = G1Projective;
-    type Uncompressed = G1Uncompressed;
-    type Compressed = G1Compressed;
+impl ConditionallySelectable for G1Projective {
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        G1Projective(blst_p1 {
+            x: Fp::conditional_select(&a.x(), &b.x(), choice).0,
+            y: Fp::conditional_select(&a.y(), &b.y(), choice).0,
+            z: Fp::conditional_select(&a.z(), &b.z(), choice).0,
+        })
+    }
+}
 
-    fn zero() -> Self {
+impl G1Affine {
+    pub fn zero() -> Self {
```

Comment:
lets unify naming around `identity` and `generator`, rather than `zero` and `one`, otherwise it will get very confusiong